### PR TITLE
move CFileItem::GetLocalFanart to ArtUtils

### PR DESCRIPTION
**Pull request by notspiff**
## Description
move CFileItem::GetLocalFanart to ArtUtils

## Motivation and context
Less code in CFileItem.

## How has this been tested?
By the added tests.

## What is the effect on users?
None

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [x] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

File: xbmc/utils/test/TestArtUtils.cpp [hunk 1]
```diff
@@ -7,19 +7,75 @@
...
+  std::string result;
+};
+
+class GetLocalFanartTest : public testing::WithParamInterface<FanartTest>, public testing::Test
+{
+};
+
+const auto local_fanart_tests = std::array{
+    FanartTest{"stack://#DIRECTORY#/foo-cd1.avi , #DIRECTORY#/foo-cd2.avi", "foo-fanart.jpg"},
+    FanartTest{"stack://#DIRECTORY#/foo-cd1.avi , #DIRECTORY#/foo-cd2.avi", "foo-cd1-fanart.jpg"},
```
Comment (thexai) [inline xbmc/utils/test/TestArtUtils.cpp:62]: Windows test failure is related to backslash but not sure why only fails the first two. Changing to this test pass (on Windows):

```
    FanartTest{"stack://#DIRECTORY#\\foo-cd1.avi , #DIRECTORY#\\foo-cd2.avi", "foo-fanart.jpg"},
    FanartTest{"stack://#DIRECTORY#\\foo-cd1.avi , #DIRECTORY#\\foo-cd2.avi", "foo-cd1-fanart.jpg"},
```

Temp dir on Windows is returned with backslash style then seems the fail is due mix slash and backslash in the path.

**Comment by notspiff** [comment]
Failure is me.doing manual path manipulation using /. @fuzzard has given me
the info rrquired. I just need the time. Thx.

**Comment by notspiff** [comment]
jenkins build this please

**Comment by notspiff** [comment]
all green now.

**Review by garbear — APPROVED** [review]
Refactor looks like it was done correctly. Good job adding tests.

**Pull request merged**